src/include/mpir_comm.h

```diff
@@ -172,6 +172,12 @@ struct MPIR_Comm {
 
     int revoked;                /* Flag to track whether the communicator
                                  * has been revoked */
+    bool uniform_ranks;         /* Flag to that checks whether there
+                                 * are same number of processes across nodes or not */
+    int max_num_procs;          /* provides processes per node, to be used when processes per
+                                 * node is same across the nodes */
+    int threshold;              /* provides threshold value to be used to select among the
+                                 * collective algorithms */
```

Comment:
How would one `threshold` value work for multiple collective functions? I understand this PR is just for `Allgather_intra_auto`, but we have to consider the scalability.

I guess we could make this into an array of size 64, one for each collective function. But for more sophisticated tuning, we may (probably) need multiple parameter threshold for each function...

Reply:
Yes, this PR was a trial run with one collective and addressing salability is still a to do work